Upload Jessica Greene's interview 2020 (#833)
rails-girls-summer-of-code/summer-of-code: blog/_posts/2020-03-18-alumni-interview-jessica.md

---
title: Alumni Interview with Jessica Greene
layout: post
created_at: Weds Mar 18 2020
permalink: blog/2020-03-18-alumni-interview-jessica
current: blog
author: RGSoC Team
twitter: RailsGirlsSoC
categories:
 - alumna-series
---
 
![](/img/blog/2020/Alumni-Interview-JessicaG.png)
 
> **“My claim to fame is making coffee for the late Christopher Lee!”**
 

> Jessica Greene has been living in Berlin for 10 years. After five years in the film & television industry and a stint as a Barista, she took the leap into coding and has since gone on to work as a back-end focused software engineer at Ecosia, the search engine that plants trees.

Reviewer: ```suggestion
Jessica Greene has been living in Berlin for 10 years. After five years in the film & television industry and a stint as a Barista, she took the leap into coding and has since gone on to work as a backend-focused software engineer at Ecosia, the search engine that plants trees.
```

 
Even if making hot beverages for Saruman himself doesn’t impress you, the following interview detailing Jessica’s time working on the <a href=" https://nextcloud.com/"> Nextcloud </a> project during our last edition of RGSoC in 2018 is bound to.
 
<h4>What first inspired you to get involved with programming? </h4>
I consider myself a self-taught/community taught developer. After leaving my job at Barn Coffee Roasters in 2017 (and having some much needed time off!) I decided to start my path to becoming a programmer. I had done a very small amount of HTML and worked with flash animations during my art foundation at Leeds art college and my partner works in games who was very supportive and encouraged me to try programming. I started with online courses, mainly web technology and then started to explore a couple of other languages such as Python and GO.
 
<h4>Did you always plan a career in tech? </h4>

> <p>Absolutely not! This is my third professional career and I have had a number of other work experiences. Changing careers is incredibly nerve-racking I was fortunate to have a supportive partner and family as well as being able to gain support from the German government. At first, I thought it would be starting from scratch and that I had nothing to take with me from my past work but that couldn't be further from the truth!</p>

Reviewer: ```suggestion
<p>Absolutely not! This is my third professional career and I have had a number of other work experiences. Changing careers is incredibly nerve-racking. I was fortunate to have a supportive partner and family as well as being able to gain support from the German government. At first, I thought it would be starting from scratch and that I had nothing to take with me from my past work but that couldn't be further from the truth!</p>
```

 
<p>Both my previous careers have fed in some obvious ways (time management/organisation) to the more abstract like documentation structure and production workflows.</p>
 
<h4>What was your RGSoC project all about?</h4>

> Nextcloud is an open source productivity platform, similar to Dropbox or Google Suite. It can be self-hosted and is focused on providing a private and secure platform where the user has control of who can access their files and where they are stored. It also has a number of apps such as a call app, calendar app, and documents app. We choose to work with the developer responsible for the contacts app which was being ported from Angular js and rebuilt using VUE.js.

Reviewer: ```suggestion
Nextcloud is an open source productivity platform, similar to Dropbox or Google Suite. It can be self-hosted and is focused on providing a private and secure platform where the user has control of who can access their files and where they are stored. It also has a number of apps such as a call app, calendar app, and documents app. We chose to work with the developer responsible for the contacts app which was being ported from Angular js and rebuilt using VUE.js.
```

 
<h4>Which skills did you find most useful during RGSoC?</h4>

> The big difference between doing RGSoC over working on my own projects was the collaboration not only with my teammate Arati but also our coaches. We tried to implement some agile practices into our workflow such as stand-ups and review sessions. I found skills from my previous job as head of production at the Barn helped. Time management and self-organisation were key and luckily these were skills I could bring with me. Patience is also a big virtue for working as a programmer, perhaps especially as a junior.

Reviewer: ```suggestion
The big difference between doing RGSoC over working on my own projects was the collaboration not only with my teammate Arati but also our coaches. We tried to implement some agile practices into our workflows such as stand-ups and review sessions. I found skills from my previous job as head of production at the Barn helped. Time management and self-organization were key and luckily these were skills I could bring with me. Patience is also a big virtue for working as a programmer, perhaps especially as a junior.
```

 
<h4>What challenges did you encounter during the program? </h4>
<p>There were a lot of things that had to be set up on our machines and working with different operating systems could be tricky at times!</p>

<p>At first, I did not have a partner for the RGSoC so I put a message in one of my local meetup groups. You have to work really closely with your partner during RGSoC and I was nervous about how this would be with someone I did not know previously.</p>
 
<p>In the end, I got on extremely well with my partner, Arati, and I struck a friendly and professional balance that supported both our individual personalities. We have remained close friends since the scholarship. Pairing with another programmer can be scary, you have to be prepared to be in a vulnerable place which can feel uncomfortable. However, the learning curve of when you work through a problem with another person is increased. Having to explain to each other what we had understood really helped cement our knowledge and we kept each other motivated. We didn't always pair but we had lots of check-ins with each other.</p>
 
<h4>What do you do in your current job role? </h4>

> <p>At Ecosia we are building services on top of the Bing API. I work in the engagement team who focused on ways we can connect users to the impact their searches are having. This past year I have been building a service to allow users to sync their personal counter value across devices. The server is built-in GO using GRPC, it's been a great opportunity to learn more about the technology while building a brand new service. I really enjoy working with the team here, especially the other junior developer.</p>

Reviewer: ```suggestion
<p>At Ecosia we are building services on top of the Bing API. I work in the engagement team who focused on ways we can connect users to the impact their searches are having. This past year I have been building a service to allow users to sync their personal counter value across devices. The server is built in GO using GRPC, it's been a great opportunity to learn more about the technology while building a brand new service. I really enjoy working with the team here, especially the other junior developer.</p>
```

 
<p>Having support within the company really motivates me to keep improving my skill set and I love hearing about the tree planting projects we are partnering with.</p>
 
<h4>How did your participation in RGSoC help you get to where you are today? </h4>

> Straight after the scholarship I started an internship at Ecosia (where I now work). Having the scholarship on my CV definitely helped prove I was capable of collaborative development work. Additionally Ecosia were just moving their front end code over to VUE.js which I had spent the scholarship working in. Having built a support network via the scholarship through our coaches and my teammate Arati, I felt really confident and ready to start my first job.

Reviewer: ```suggestion
Straight after the scholarship I started an internship at Ecosia (where I now work). Having the scholarship on my CV definitely helped prove I was capable of collaborative development work. Additionally, Ecosia was just moving their front end code over to VUE.js which I had spent the scholarship working in. Having built a support network via the scholarship through our coaches and my teammate Arati, I felt really confident and ready to start my first job.
```

 
<h4>Do you still do any programming? </h4>
Outside of programing at work I am involved in a couple of local community groups including PyLadies, Women Who GO and Open Source Diversity. I also have a couple of side projects for my personal learning. It's definitely important to have a balance and I try to make sure I also have offline time where I can. Then I love to read, go to the sauna and go for walks.
 
<h4>Do you have any advice for new RGSoC students and for women and non-binary people who wish to work in tech? </h4>

> Persevere! It is nerve-racking to get into any new industry and it is different for everyone. I recognise I have been extremely fortunate and privileged to live somewhere where there is a vivid tech industry with many companies looking to employ developers and many meetup groups to support underrepresented and marginalised groups. Community has been core to my success, not only the shared learning experience but also the power to stay motivated and keep going. I really recommend, be it online or in person, finding people in a similar place to you with who you can share successes and frustrations.

Reviewer: ```suggestion
Persevere! It is nerve-racking to get into any new industry and it is different for everyone. I recognise I have been extremely fortunate and privileged to live somewhere where there is a vivid tech industry with many companies looking to employ developers and many meetup groups to support underrepresented and marginalised groups. Community has been core to my success, not only the shared learning experience but also the power to stay motivated and keep going. I really recommend, be it online or in person, finding people in a similar place to you with whom you can share successes and frustrations.
```
